index.d.ts

```diff
@@ -626,3 +626,49 @@ interface KVNamespace {
     cursor: string;
   }>;
 }
+
+interface DurableObjectEntries<T> {
+  [key: string]: T;
+}
+
+interface DurableObjectListOptions {
+  start: string;
+  end: string;
+  reverse: boolean;
+  limit: number;
+}
+
+interface DurableObjectStorage {
+  get<ExpectedValue = unknown>(key: string): Promise<ExpectedValue>;
+  get<ExpectedValue = unknown>(keys: Array<string>): Promise<Map<string, ExpectedValue>>;
+  put<GivenValue = unknown>(key: string, value: GivenValue): Promise<void>;
+  put<GivenValue = unknown>(entries: DurableObjectEntries<GivenValue>): Promise<void>;
+  delete(key: string): Promise<boolean>;
+  delete(keys: Array<string>): Promise<boolean>;
+  list<ExpectedValue = unknown>(
+    options?: DurableObjectListOptions
+  ): Promise<Map<string, ExpectedValue>>;
+  transaction<ExpectedValue = unknown>(
+    closure: (txn: DurableObjectStorage) => Promise<void>
+  ): Promise<Map<string, ExpectedValue>>;
+}
+
+interface DurableObjectState {
+  storage: DurableObjectStorage;
+}
+
+interface DurableObject {
```

Comment:
I think there's probably going to be some confusion between the durable object class, and the stub used to send messages to the durable object.  Maybe call this DurableObjectStub?